_posts/2018-03-22-industrialization-versus-data-selection.md

```diff
@@ -0,0 +1,47 @@
+Historically, our LMs were crafted one by one with love, with a nice up of human intervention in between. Meaning that we had to experiment to find the best system with our own eyes. And this is so not compatible with automating.
+What is data selection you may ask? However, first thing first.
+
+## ASR: Automatic Speech Recognition Prelude
+
+In ASR, we always consider building two independent modules which will be mixed together later on. Each module in itself is very dependent of the language we want to recognize.
+
+- The first one is called acoustic model. It represents the way of speaking. What sounds can be put together to form a word, a sentence. In fact, we represent the language by a series of phoneme. If we watch in [Wiktionary](https://en.wiktionary.org/wiki/phoneme) it's clearer isn't it? Don't confuse with syllable through. We can take an example. The word 'through'(1 syllable) consists of three sounds, three phonemes: 'th' 'r' 'oo'. So the aim is to models the sounds as a sequence of phonemes.
+- The second one is the language model, which we want to build automatically. It models the distribution of the words for a given language. Thanks to those probabilities, the LM helps picking the best correspondence between a sequence of phonemes and the words/sentences.
+
+In order to build these modules, we need data and the more we have and the more they are relevant, the better! That's why we need data selection: we need a mean to retrieve relevant data adapted to the context of the recognition. In fact a lawyer and a baker don't speak the same language: they don't use the same lexicon. The data selection is picking the good data that match the domain within millions of examples through the usage of various automatic algorithms.
+
+## How we used to do
+
+As we discussed above, data selection is a very important step while building a system. Like many, we used the [Moore-Lewis method](http://www.aclweb.org/anthology/P10-2041) which was also adapted for bilingual use (like translation) by Axelrod et al. in [Domain Adaptation via Pseudo In-Domain Data Selection](https://aclanthology.info/pdf/D/D11/D11-1033.pdf). These are very effective ways to select data using two corpora (in and out domain) by comparing cross-entropies. In domain meaning that the corpus have specific data, that are relevant with the context, the domain of the recognition as explained before with the lawyer/baker thing. Whereas out domain is just a pool of random data, meaning there is relevant and no-relevant data in it! Then about the cross-entropy, it's the method that help choosing the well-matched data for the desired output. Thanks to some relevant segment, we compare each segment in the pool of data to retrieve the closest to the initial data.
```

Comment:
I'm not sure I get this sentence : "In domain meaning that the corpus have specific data, that are relevant with the context, the domain of the recognition as explained before with the lawyer/baker thing."

Reply:
Oh ! ok, I should ready "in domain" as one expression, it's not the preposition "in". I understood thanks to following sentence "whereas out domain". Maybe use quotes, bold or italic to distinguish it?

Reply:
As Simon says (oh! oh! oh!), you should use a marker for "in domain" and "out of domain"... I suggest "in-domain" and "out-of-domain" (that's what is used among the community AFAIK).

Reply:
In "Moore-Lewis method", method shouldn't be part of the link

Reply:
"the domain of the recognition" => remove "the"
"about the cross-entropy" => remove "the"
"the well-matched data" => remove "the"
General comment: you use too much "the" in your flow!

Reply:
"the closest to the" => "the closest ones to the"

Reply:
Cross-entropy is not a method, it is a measure